conans/model/new_build_info.py

```diff
@@ -47,21 +47,21 @@ def required_component_names(self):
             return []
         return [r for r in self.requires if "::" not in r]
 
-    def set_property(self, property_name, value, generator=None):
+    def set_property(self, property_name, value, generator=None, path=False):
```

Comment:
I would prefer `is_path`, but anyway, why is not the consumer of the property the one "adapting" the values?

Reply:
Because that is a big challenge. 
The consumers receive all the ``includedirs`` and ``properties`` already merged. When the installer processes the ``layout()`` and take into accounts editables, the ``cpp.source`` and ``cpp.build`` are merged into the final ``cpp.package`` (or ``cpp_info``). If we want the consumer to be able to handle that we would need:
- To add to every single item (every dir in ``includedirs``, ``libdirs``, and every property), the information if it is a ``source``, ``build`` or ``package`` item
- To make accessible to the consumer the final ``build_folder``, ``source_folder``, and ``package_folder`` (the later is always available, but the others are not for normal packages, they are set to None in ``installer``, because the build folder and source folder don't even exist, unless they are editable
- To let the consumer know which packages are in editable mode.
- Then, every consumer needs to be composing with ``set_relative_folder``, passing the right folder depending if it is editable or not, and using the right folder depending on the origin. If this functionality can be wrapped in a single call for convenience (a call that must be called by every generator in order to work), then it makes no sense to let the consumers call it, and not have it called automatically by Conan, so it is already processed when it gets to the generators.

Reply:
I see. I'm afraid we are digging the hole again. Anyway, please, call it `is_path`